### PR TITLE
Move CredentialsBindingHandler extension point to extensions plugin

### PR DESCRIPTION
**Pull request by abayer**
* JENKINS issue(s):
    * n/a
* Description:
    * This way, plugins will be able to just depend on pipeline-model-extensions if they want to implement a handler. I also moved the actual implementations out of CredentialsBindingHandler and into a better package - I couldn't move CredentialsBindingHandler itself into a better package because I didn't want to break compatibility.
* Documentation changes:
    * n/a - internal relocation only
* Users/aliases to notify:
    * @reviewbybees

### CODE REVIEW

File: pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/credentials/impl/FileCredentialsHandler.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018, CloudBees, Inc.
```
Comment (rsandell) [inline pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/credentials/impl/FileCredentialsHandler.java]: 2016 - 2018

File: pipeline-model-extensions/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/model/CredentialsBindingHandler.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /*
  * The MIT License
  *
- * Copyright (c) 2016, CloudBees, Inc.
+ * Copyright (c) 2018, CloudBees, Inc.
```
Comment (rsandell) [inline pipeline-model-extensions/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/model/CredentialsBindingHandler.java]: 2016 - 2018

File: pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/credentials/impl/StringCredentialsHandler.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    public Class<? extends StandardCredentials> type() {
+        return StringCredentials.class;
+    }
+
+    @Nonnull
+    @Override
+    public List<Map<String, Object>> getWithCredentialsParameters(String credentialsId) {
+        Map<String, Object> map = new HashMap<>();
+        map.put("$class", StringBinding.class.getName());
+        map.put("variable", new EnvVarResolver());
```
Comment (rsandell) [inline pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/credentials/impl/StringCredentialsHandler.java]: What `EnvVarResolver` are these classes picking up? The original was a static internal class. I didn't think those were inherited?

Comment (abayer) [reply pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/credentials/impl/StringCredentialsHandler.java]: They are inherited, apparently. I guess we could move `EnvVarResolver` to its own file as well for clarity.

Comment (rsandell) [reply pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/credentials/impl/StringCredentialsHandler.java]: Unless someone has started using it already?

Comment (abayer) [reply pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/credentials/impl/StringCredentialsHandler.java]: Yeah, better safe than sorry - let's leave it as is.

**Pull request merged**